packages/sanity/src/core/form/studio/tree-editing/utils/build-tree-editing-state/utils.ts

```diff
@@ -0,0 +1,34 @@
+import {type Path} from 'sanity'
+
+/**
+ * Check if the item is selected
+ */
+export function isSelected(itemPath: Path, path: Path): boolean {
+  return JSON.stringify(itemPath) === JSON.stringify(path)
+}
+
+/**
+ * Check if the path is an array item path
+ */
+export function isArrayItemPath(path: Path): boolean {
+  return path[path.length - 1].hasOwnProperty('_key')
+}
+
+/**
+ * Check if the item should be in the breadcrumb
+ */
+export function shouldBeInBreadcrumb(itemPath: Path, path: Path): boolean {
+  return (
+    itemPath.every((segment, index) => {
+      return JSON.stringify(path[index]) === JSON.stringify(segment)
```

Comment:
thinking out loud, wouldn't this work in theory. Although the types might be different?

```suggestion
      return isSelected(path[index], segment)
```

Reply:
This does sound about right!
I'll try it out, see if any of the tests screams or if something is iffy 👍